File: FestivalC#/Festival/Festival/Validation/ValidatorShow.cs
using Festival.Model;
using Festival.Validation.Exceptions;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Festival.Validation
{
    public class ValidatorShow : IValidator<Show>
    {
        public void validate(Show show)
        {
            String errors = "";
            if (show.Location.Length == 0)
            {
                errors += "Location must not be null";
            }
            if (show.TicketsAvailable < 0)
            {
                errors += "Availbale tickets must be a positive number";
            }
            if (show.TicketsSold < 0)
            {
                errors += "Sold tickets must be a positive number";
            }
            if (errors.Length != 0)
            {
                throw new ValidatorException(errors);
            }
        }
    }
}

File: FestivalJava/Services/src/main/java/Validation/ValidatorShow.java
package Validation;

import Domain.Show;
import Validation.Exceptions.ValidatorException;

/**
 * Created by Sebi on 18-Mar-17.
 */
public class ValidatorShow implements IValidator<Show> {
    @Override
    public void validate(Show show) throws ValidatorException {
        String errors = "";
        if (show.getLocation().length() == 0){
            errors += "Location must not be null";
        }
        if (show.getDate().length() == 0){
            errors += "Date must not be null";
        }
        if (show.getTicketsAvailable() < 0){
            errors += "Availbale tickets must be a positive number";
        }
        if (show.getTicketsSold() < 0){
            errors += "Sold tickets must be a positive number";
        }
        if (errors.length() != 0){
            throw new ValidatorException(errors);
        }
    }
}
